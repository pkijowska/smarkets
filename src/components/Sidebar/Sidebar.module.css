.wrapper {
    margin: 0 20px;
    font-weight: 500;
    background: var(--background-grey);
    padding: 10px;
    width: 300px;
    border-radius: 10px;
    text-align: center;
}

.header {
    color: var(--primary-green);
    margin-bottom: 20px;
    font-weight: 800;
    font-size: larger;
}

.wrapper ul {
    color: var(--color-white);
    display: flex;
    flex-direction: column;
}

.wrapper ul li {
    margin-bottom: 20px;
}

.active {
    text-decoration: underline;
    text-decoration-color: var(--primary-green); 
    text-decoration-thickness: 4px;
}