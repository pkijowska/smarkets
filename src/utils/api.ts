import { Events, EventDetails } from "@/types/types";

export async function getEvents(category: string): Promise<Events> {
    try {
      const response = await fetch(
        `https://api.smarkets.com/v3/events/?state=upcoming&type=${category}_match&type_domain=${category}&type_scope=single_event&with_new_type=false&sort=id&limit=20&include_hidden=false`
      );
  
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      const data = await response.json();
      return data;
    } catch (error) {
      throw new Error("Error during fetch: " + (error as Error).message);
    }
}

export const getEventDetails = async (id: string): Promise<EventDetails> => {
    try {
      const response = await fetch(
        `https://api.smarkets.com/v3/events/${id}/states/`
      );
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      const data = await response.json();
      return data;
    } catch (error) {
      throw new Error("Error during fetch: " + (error as Error).message);
    }
};