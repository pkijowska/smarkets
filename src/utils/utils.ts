import { Categories } from "../types/types";

export const CATEGORIES:Categories = [
    {  label: 'Football', slug: 'football', emoji: '⚽️' },
    { label: 'Tennis', slug: 'tennis', emoji: '🎾' },
    {  label: 'Basketball', slug: 'basketball', emoji: '🏀' },
    { label: 'Baseball', slug: 'baseball', emoji: '⚾️' },
    {  label: 'Rugby Union', slug: 'rugby_union', emoji: '🏉' },
    {  label: 'Boxing', slug: 'boxing', emoji: '🥊' },
    {  label: 'Cricket', slug: 'cricket', emoji: '🏏' },
];

export function formatUTCDate(utcDateString: string) {
    const date = new Date(utcDateString);
    const day = date.getUTCDate().toString().padStart(2, '0');
    const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    const month = months[date.getUTCMonth()];
    const year = date.getUTCFullYear();
    const hours = date.getUTCHours().toString().padStart(2, '0');
    const minutes = date.getUTCMinutes().toString().padStart(2, '0');
    return `${day} ${month} ${year} ${hours}:${minutes} GMT`;
}