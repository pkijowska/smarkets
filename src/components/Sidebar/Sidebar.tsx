import { CATEGORIES } from "@/utils/utils";
import Link from "next/link";
import styles from "./Sidebar.module.css";

const Sidebar = ({ selectedCategory }: { selectedCategory: string }) => {
  return (
    <aside className={styles.wrapper}>
      <nav>
        <h2 className={styles.header}>Categories</h2>
        <ul>
          {CATEGORIES.map(({ slug, label, emoji }) => {
            return (
              <li key={slug}>
                <Link
                  href={`/events/${slug}`}
                  className={label === selectedCategory ? styles.active : ""}
                  data-testid={
                    label === selectedCategory ? "active" : "inactive"
                  }
                >
                  <span
                    aria-hidden="true"
                    className={label === selectedCategory ? "" : styles.emoji}
                  >
                    {emoji}
                  </span>{" "}
                  {label}
                </Link>
              </li>
            );
          })}
        </ul>
      </nav>
    </aside>
  );
};
export default Sidebar;
