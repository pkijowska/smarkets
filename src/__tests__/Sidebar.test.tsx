import { render } from "@testing-library/react";
import { CATEGORIES } from "@/utils/utils";
import Sidebar from "@/components/Sidebar/Sidebar";

describe("CategorySidebar", () => {
  it("matches the snapshot", () => {
    const { container } = render(<Sidebar selectedCategory={"Football"} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the 7 category links", () => {
    render(<Sidebar selectedCategory={"football"} />);
    const links = CATEGORIES.length;
    expect(links).toBe(7);
  });

  it("highlights the selected category", () => {
    const { container } = render(<Sidebar selectedCategory={"Football"} />);
    const selected = container.querySelector(
      "a[data-testid='active']"
    )?.textContent;

    expect(selected).toBe("⚽️ Football");
  });
});
