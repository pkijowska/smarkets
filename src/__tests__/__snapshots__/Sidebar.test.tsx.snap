// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySidebar matches the snapshot 1`] = `
<div>
  <aside
    class="wrapper"
  >
    <nav>
      <h2
        class="header"
      >
        Categories
      </h2>
      <ul>
        <li>
          <a
            class="active"
            data-testid="active"
            href="/events/football"
          >
            <span
              aria-hidden="true"
              class=""
            >
              ⚽️
            </span>
             
            Football
          </a>
        </li>
        <li>
          <a
            class=""
            data-testid="inactive"
            href="/events/tennis"
          >
            <span
              aria-hidden="true"
              class="emoji"
            >
              🎾
            </span>
             
            Tennis
          </a>
        </li>
        <li>
          <a
            class=""
            data-testid="inactive"
            href="/events/basketball"
          >
            <span
              aria-hidden="true"
              class="emoji"
            >
              🏀
            </span>
             
            Basketball
          </a>
        </li>
        <li>
          <a
            class=""
            data-testid="inactive"
            href="/events/baseball"
          >
            <span
              aria-hidden="true"
              class="emoji"
            >
              ⚾️
            </span>
             
            Baseball
          </a>
        </li>
        <li>
          <a
            class=""
            data-testid="inactive"
            href="/events/rugby_union"
          >
            <span
              aria-hidden="true"
              class="emoji"
            >
              🏉
            </span>
             
            Rugby Union
          </a>
        </li>
        <li>
          <a
            class=""
            data-testid="inactive"
            href="/events/boxing"
          >
            <span
              aria-hidden="true"
              class="emoji"
            >
              🥊
            </span>
             
            Boxing
          </a>
        </li>
        <li>
          <a
            class=""
            data-testid="inactive"
            href="/events/cricket"
          >
            <span
              aria-hidden="true"
              class="emoji"
            >
              🏏
            </span>
             
            Cricket
          </a>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
